prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -200,6 +203,32 @@ object CustomFHIRFunctions {
             else -> mutableListOf()
         }
     }
+
+    /**
+     * Gets the FHIR gender stored in the [focus] element
+     * and converts to HL7 v2.5.1 - 0001 - Administrative Sex
+     * @return a mutable list containing the HL7 single character version of the code
+     */
+    fun getAdministrativeGenderCode(focus: MutableList<Base>): MutableList<Base> {
+        return when (AdministrativeGender.fromCode((focus[0] as Enumeration<*>).code)) {
+            AdministrativeGender.UNKNOWN -> mutableListOf(StringType("U"))
+            AdministrativeGender.FEMALE -> mutableListOf(StringType("F"))
+            AdministrativeGender.MALE -> mutableListOf(StringType("M"))
+            AdministrativeGender.OTHER -> mutableListOf(StringType("O"))
+            else -> mutableListOf()
+        }
+    }
+
+    /**
+     * Translate a boolean-type value into the single character Y/N value used by HL7.
+     * @return a mutable list containing the HL7 single character version of the code
+     */
+    fun getYesNoValue(focus: MutableList<Base>): MutableList<Base> {
+        return when (focus[0].toString()) {
+            "true" -> mutableListOf(StringType("Y"))
+            else -> mutableListOf(StringType("N"))
```

Comment:
else needs to be empty string.  You do not want to add data to a field that does not have a value

Reply:
Fixed